it/uk/gov/hmrc/soletraderidentificationfrontend/repositories/JourneyConfigRepositoryISpec.scala

```diff
@@ -0,0 +1,80 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.soletraderidentificationfrontend.repositories
+
+import org.scalatest.concurrent.{AbstractPatienceConfiguration, Eventually}
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.Helpers._
+import play.api.{Application, Environment, Mode}
+import uk.gov.hmrc.soletraderidentificationfrontend.assets.TestConstants._
+import uk.gov.hmrc.soletraderidentificationfrontend.utils.ComponentSpecHelper
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class JourneyConfigRepositoryISpec extends ComponentSpecHelper with AbstractPatienceConfiguration with Eventually {
+
+  override lazy val app: Application = new GuiceApplicationBuilder()
+    .in(Environment.simple(mode = Mode.Dev))
+    .configure(config)
+    .configure("application.router" -> "testOnlyDoNotUseInAppConf.Routes")
+    .configure("mongodb.timeToLiveSeconds" -> "10")
+    .build
+
+  val repo: JourneyConfigRepository = app.injector.instanceOf[JourneyConfigRepository]
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    await(repo.drop)
+  }
+
+  "documents" should {
+    "successfully insert a new document" in {
+      await(repo.insertJourneyConfig(testJourneyId, testInternalId, testIndividualJourneyConfig))
+      await(repo.count) mustBe 1
+    }
+
+    "successfully insert journeyConfig" in {
+      await(repo.insertJourneyConfig(testJourneyId, testInternalId, testIndividualJourneyConfig))
+      await(repo.findJourneyConfig(testJourneyId, testInternalId)) must contain(testIndividualJourneyConfig)
+    }
+
+    "successfully delete all documents" in {
+      await(repo.insertJourneyConfig(testJourneyId, testInternalId, testIndividualJourneyConfig))
+      await(repo.drop)
+      await(repo.count) mustBe 0
+    }
+
+    "successfully delete one document" in {
+      await(repo.insertJourneyConfig(testJourneyId, testInternalId, testIndividualJourneyConfig))
+      await(repo.insertJourneyConfig(testJourneyId + 1, testInternalId, testIndividualJourneyConfig))
+      await(repo.remove("_id" -> (testJourneyId + 1), "authInternalId" -> testInternalId))
+      await(repo.count) mustBe 1
+    }
+
+    "successfully find journey config" when {
+      "regime key is present" in {
+        await(repo.insertJourneyConfig(testJourneyId, testInternalId, testIndividualJourneyConfig))
+        await(repo.findJourneyConfig(testJourneyId, testInternalId)) must contain(testIndividualJourneyConfig)
+      }
```

Comment:
I'm surprised this is working. If "removeById" is really deleting the regime property the subsequent retrieval should fail because the current "reads" implementation expects the regime to be present (if you have already updated the read try using a journey with a regime of PPT and see if it comes back with the default VATC).

If you look at the current version of IEI FE you can see how to update the database using "findAndUpdate". Try updating the database with a journey config without a regime value and see if the read fails.

Reply:
done